3scale/porta
# Bump the version of @babel/helper-create-regexp-features-plugin

**What this PR does / why we need it**:

The latest builds fail with the following error (one per each JS pack):

```
ERROR in ./app/javascript/packs/active_docs.ts
Module build failed (from ./node_modules/babel-loader/lib/index.js):
TypeError: /opt/system/app/javascript/packs/active_docs.ts: Invalid Version: 70001900000
    at new SemVer (/opt/system/node_modules/semver/semver.js:314:11)
    at compare (/opt/system/node_modules/semver/semver.js:647:10)
    at Function.lt (/opt/system/node_modules/semver/semver.js:688:10)
    at PluginPass.pre (/opt/system/node_modules/@babel/plugin-transform-named-capturing-groups-regex/node_modules/@babel/helper-create-regexp-features-plugin/lib/index.js:55:44)
    at transformFile (/opt/system/node_modules/@babel/core/lib/transformation/index.js:73:27)
    at transformFile.next (<anonymous>)
    at run (/opt/system/node_modules/@babel/core/lib/transformation/index.js:24:12)
    at run.next (<anonymous>)
    at transform (/opt/system/node_modules/@babel/core/lib/transform.js:22:41)
    at transform.next (<anonymous>)
    at step (/opt/system/node_modules/gensync/index.js:261:32)
    at /opt/system/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/opt/system/node_modules/gensync/index.js:223:11)
    at /opt/system/node_modules/gensync/index.js:189:28
    at /opt/system/node_modules/@babel/core/lib/gensync-utils/async.js:68:7
    at /opt/system/node_modules/gensync/index.js:113:33
    at step (/opt/system/node_modules/gensync/index.js:287:14)
    at /opt/system/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/opt/system/node_modules/gensync/index.js:223:11)
```

This started happening after upgrading Babel: https://github.com/3scale/porta/pull/3340

The issues seems to be a bug in Babel introduced [here](https://github.com/babel/babel/pull/15548). The issue was [reported](https://github.com/babel/babel/issues/15604) and [fixed](https://github.com/babel/babel/pull/15605) in [v7.21.8](https://github.com/babel/babel/blob/main/CHANGELOG.md#v7218-2023-05-02).

This version of `@babel/core` [was already used](https://github.com/3scale/porta/pull/3340/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R18) in latest porta, but the package `@babel/helper-create-regexp-features-plugin` which is the one that has the fix, was still resolved to an older version.

This PR just updates the `yarn.lock` file.

Generated as follows:
```
yarn upgrade --recursive @babel/helper-create-regexp-features-plugin
npx yarn-deduplicate
```

**Which issue(s) this PR fixes** 

_-none-_

**Verification steps** 

Build the image and make sure the process completes successfully:

```
podman build -f openshift/system/Dockerfile --tag porta-image .
```

**Special notes for your reviewer**:

@josemigallas had this issue before, but it looked like it was resolved by adding `.browserlistrc` (see https://stackoverflow.com/questions/76162569/babel-error-invalid-version-70001900000/76172109). I think it was probably just a coincidence, as it doesn't seem related.

This version bump also upgraded `eslint-plugin-react` from `7.31.7` to `7.32.2` which added new rules, e.g. `react/no-object-type-as-default-prop`, and some issues reported by this rule are also fixed in this PR.
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/CHANGELOG.md#added-1
